arbnode/dataposter/externalsignertest/externalsignertest.go

```diff
@@ -94,8 +98,14 @@ func NewServer(t *testing.T) (*http.Server, *SignerAPI) {
 	pool := x509.NewCertPool()
 	pool.AppendCertsFromPEM(clientCert)
 
+	ln, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Error creating listener: %v", err)
+	}
+	port := strings.Split(ln.Addr().String(), ":")[1]
+
 	httpServer := &http.Server{
-		Addr:              fmt.Sprintf(":%d", SignerPort),
+		Addr:              fmt.Sprintf(":%s", port),
```

Comment:
```suggestion
		Addr:              ln.Addr().String(),
```

You can avoid deconstructing the address since it is only used here.

Reply:
Done.